### PR TITLE
PMM-7 refactoring

### PR DESCRIPTION
**Pull request by nikita-b**

### CODE REVIEW

File: packer/ansible/agent.yml [hunk 1]
```diff
@@ -42,69 +53,31 @@
           - make
           - perl
           - p7zip
+          - https://rpm.nodesource.com/pub_14.x/el/7/{{ ansible_architecture }}/nodejs-{{ node_version }}-1nodesource.{{ ansible_architecture }}.rpm
```
Comment (atymchuk) [inline packer/ansible/agent.yml:56]: I looks we already use NodeJS v16 - https://github.com/Percona-Lab/pmm-submodules/blob/b79a9d61a56c55d4ac6fbd1261bb6837e13d2942/build/rpmbuild-docker/Dockerfile#L4

Comment (atymchuk) [reply packer/ansible/agent.yml:56]: https://github.com/percona-platform/saas-ui/blob/main/package.json#L117

File: packer/ansible/agent.yml [hunk 2]
```diff
@@ -161,6 +148,13 @@
         append: yes
       when: ansible_facts['distribution'] == "Amazon"
 
+    - name: pull and cache docker images
```
Comment (atymchuk) [inline packer/ansible/agent.yml:151]:
```suggestion
    - name: Pull and cache docker images
```

File: packer/ansible/agent.yml [hunk 3]
```diff
@@ -161,6 +148,13 @@
...
+      docker_image:
+        name: "{{ item }}"
+        source: pull
+      loop:
```
Comment (atymchuk) [inline packer/ansible/agent.yml:155]: Just one item, do we need to loop?

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: packer/ansible/agent.yml (hunk 3 above)
Comment (nikita-b) [reply packer/ansible/agent.yml:155]: Yes, I plan to add more images :)

File: packer/ansible/agent.yml [hunk 4]
```diff
@@ -42,69 +53,31 @@
...
+        dest: /usr/local/bin/docker-compose
+        mode: 'u+x,g+x,o+x'
+
+    - name: Install kubectl
+      get_url:
+        url: https://storage.googleapis.com/kubernetes-release/release/v{{ kubectl_version }}/bin/linux/{{ ansible_architecture_alt }}/kubectl
+        dest: /usr/local/bin/kubectl
+        mode: 'u+x,g+x,o+x'
+
+    - name: Install NodeJS
```
Comment (atymchuk) [inline packer/ansible/agent.yml:71]:
```suggestion
    - name: Install some software
```

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Review by atymchuk — APPROVED** [review]

**Pull request merged**